src/NzbDrone.Common/Disk/DiskTransferService.cs

```diff
@@ -0,0 +1,267 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using NLog;
+using NzbDrone.Common.EnsureThat;
+using NzbDrone.Common.Extensions;
+
+namespace NzbDrone.Common.Disk
+{
+    public interface IDiskTransferService
+    {
+        TransferMode TransferFolder(String sourcePath, String targetPath, TransferMode mode);
+        TransferMode TransferFolderVerified(String sourcePath, String targetPath, TransferMode mode);
+        TransferMode TransferFile(String sourcePath, String targetPath, TransferMode mode, bool overwrite = false);
+        TransferMode TransferFileVerified(String sourcePath, String targetPath, TransferMode mode);
+    }
+
+    public class DiskTransferService : IDiskTransferService
```

Comment:
It seems like putting the `Verification` behind a flag would help. 

`TransferMode TransferFile(String sourcePath, String targetPath, TransferMode mode, bool overwrite = false, bool verified = false);`